warehouse/warehouse.go

```diff
@@ -764,10 +649,17 @@ func (wh *HandleT) createJobs(ctx context.Context, warehouse warehouseutils.Ware
 	})
 	defer uploadJobCreationStat.RecordDuration()()
 
+	creationStart := time.Now()
+
 	uploadStartAfter := getUploadStartAfterTime()
-	wh.createUploadJobsFromStagingFiles(warehouse, whManager, stagingFilesList, priority, uploadStartAfter)
+	err = wh.createUploadJobsFromStagingFiles(ctx, warehouse, stagingFilesList, priority, uploadStartAfter)
+	if err != nil {
+		return err
+	}
 	setLastProcessedMarker(warehouse, uploadStartAfter)
 
+	uploadJobCreationStat.SendTiming(time.Since(creationStart))
```

Comment:
We are already measuring it on line 650. Do we still need this?
`defer uploadJobCreationStat.RecordDuration()()`

Reply:
added from a recent merge